smart-contracts/package.json

```diff
@@ -34,12 +34,12 @@
   },
   "scripts": {
     "remix": "remixd -s contracts --remix-ide http://localhost:8080 --read-only",
-    "test": "cross-env NODE_ENV=test truffle test --network development",
+    "test": "cross-env NODE_ENV=test truffle migrate && truffle test",
     "lint": "solhint \"contracts/**/*.sol\" && solium -d ./contracts/ && eslint .",
     "dev": "npm run lint && npm run build && npm run test",
     "ci": "npm run lint && npm run test && npm run reports",
     "build": "truffle compile --all",
-    "deploy": "truffle deploy --reset",
```

Comment:
This is the workaround we put in to work around an `out of gas` error on the second run of `unlock-app`.  It should not be necessary.

Reply:
Hmm. So this forces All migrations scripts to run each time, which shouldn't matter for dev, but definitely not the behaviour we want on Ropsten or mainnet.

Reply:
Right.  This PR aims to fix that.